crud/upsert_many.go

```diff
@@ -0,0 +1,104 @@
+package crud
+
+import (
+	"context"
+
+	"github.com/tarantool/go-tarantool"
+)
+
+type UpsertManyOpts = OperationManyOpts
+
+// UpsertManyRequest helps you to create request object to call
+// `crud.upsert_many` for execution by a Connection.
+type UpsertManyRequest struct {
+	spaceRequest
+	tuplesOperationData interface{}
+	opts                UpsertManyOpts
+}
+
+// NewUpsertManyRequest returns a new empty UpsertManyRequest.
+func NewUpsertManyRequest(space interface{}) *UpsertManyRequest {
+	req := new(UpsertManyRequest)
+	req.initImpl("crud.upsert_many")
+	req.setSpace(space)
+	req.tuplesOperationData = []interface{}{}
+	req.opts = UpsertManyOpts{}
+	return req
+}
+
+// TuplesOperationData sets tuples and operations for
+// the UpsertManyRequest request.
+// Note: default value is nil.
+func (req *UpsertManyRequest) TuplesOperationData(tuplesOperationData interface{}) *UpsertManyRequest {
+	req.tuplesOperationData = tuplesOperationData
+	return req
+}
+
+// Opts sets the options for the UpsertManyRequest request.
+// Note: default value is nil.
+func (req *UpsertManyRequest) Opts(opts UpsertManyOpts) *UpsertManyRequest {
+	req.opts = opts
+	return req
+}
+
+// Body fills an encoder with the call request body.
+func (req *UpsertManyRequest) Body(res tarantool.SchemaResolver, enc *encoder) error {
+	req.impl.Args([]interface{}{req.space, req.tuplesOperationData, convertToMap(req.opts)})
+	return req.impl.Body(res, enc)
+}
+
+// Context sets a passed context to CRUD request.
+func (req *UpsertManyRequest) Context(ctx context.Context) *UpsertManyRequest {
+	req.impl = req.impl.Context(ctx)
+
+	return req
+}
+
+type UpsertObjectManyOpts = OperationManyOpts
+
+// UpsertObjectManyRequest helps you to create request object to call
+// `crud.upsert_object_many` for execution by a Connection.
+type UpsertObjectManyRequest struct {
+	spaceRequest
+	objectsOperationData interface{}
+	opts                 UpsertObjectManyOpts
+}
+
+// NewUpsertObjectManyRequest returns a new empty UpsertObjectManyRequest.
+func NewUpsertObjectManyRequest(space interface{}) *UpsertObjectManyRequest {
```

Comment:
Why space is an `interface{}`? CRUD expects `string` space, with the only exception of `crud.len(space_id)`, which has beed deprecated since `0.14.0` and removed in `1.0.0` (it's better to ignore it rather than support, see https://github.com/tarantool/crud/issues/255).

Reply:
Fixed.